lib/internal/Magento/Framework/Image/Adapter/Gd2.php

```diff
@@ -778,4 +778,68 @@ protected function _createEmptyImage($width, $height)
         $this->imageDestroy();
         $this->_imageHandler = $image;
     }
+
+    /**
+     * Fix an issue with the usage of imagecopymerge where the alpha channel is lost
+     *
+     * @param resource $dst_im
+     * @param resource $src_im
+     * @param int $dst_x
+     * @param int $dst_y
+     * @param int $src_x
+     * @param int $src_y
+     * @param int $src_w
+     * @param int $src_h
+     * @param int $pct
+     *
+     * @return bool
+     */
+    private function imagecopymergeWithAlphaFix(
+        $dst_im,
+        $src_im,
+        $dst_x,
+        $dst_y,
+        $src_x,
+        $src_y,
+        $src_w,
+        $src_h,
+        $pct
+    ) {
+        if ($pct >= 100) {
+            return imagecopy($dst_im, $src_im, $dst_x, $dst_y, $src_x, $src_y, $src_w, $src_h);
+        }
+
+        if ($pct < 0) {
+            return false;
+        }
+
+        $sizeX = imagesx($src_im);
+        $sizeY = imagesy($src_im);
+        if (false === $sizeX || false === $sizeY) {
```

Comment:
Please convert all Yoda-style conditions to normal ones.

Is checking for `false` really necessary here - do we need to continue execution flow when `$sizeX === 0`?